packtools/sps/validation/article_toc_sections.py

```diff
@@ -2,58 +2,306 @@
-            if text in (expected_toc_sections.get(lang) or []):
-                message = "OK, section titles match the document"
-                result = True
+        """
+        Check whether the TOC sections matches the options provided in a standard list.
+
+        XML input
+        ---------
+        <article xml:lang="es">
+        <front>
+            <article-meta>
+                <title-group>
+                    <article-title>Título do artigo em espanhol</article-title>
+                </title-group>
+                <article-categories>
+                    <subj-group subj-group-type="heading">
+                        <subject>Nome da seção do artigo em espanhol</subject>
+                    </subj-group>
+                </article-categories>
+            </article-meta>
+        </front>
+        <sub-article article-type="translation" xml:lang="en">
+            <front-stub>
+                <article-categories>
+                    <subj-group subj-group-type="heading">
+                        <subject>Nome da seção do sub-artigo em inglês</subject>
+                    </subj-group>
+                </article-categories>
+                <title-group>
+                    <article-title>HEDGING FUTURE CASH FLOWS WITH INTEREST-RATE FUTURES CONTRACTS: A DURATION AND
+                     CONVEXITY ANALYSIS UNDER THE NELSON SIEGEL MODEL</article-title>
+                </title-group>
+            </front-stub>
+        </sub-article>
+        </article>
+
+        Params
+        ------
+        expected_toc_sections : dict, such as:
+            {
+                "es": ["Nome da seção do artigo em espanhol"],
+                "en": ["Nome da seção do sub-artigo em inglês"]
+            }
+
+        Returns
+        -------
+        list of dict
+            A list of dictionaries, such as:
+            [
+                {
+                'title': 'Article section title validation',
+                'xpath': ".//article-meta//subj-group[@subj-group-type='heading']/subject",
+                'validation_type': 'match',
+                'response': 'OK',
+                'expected_value': ['Nome da seção do artigo em espanhol'],
+                'got_value': ['Nome da seção do artigo em espanhol'],
+                'message': 'Got [\'Nome da seção do artigo em espanhol\'] expected [\'Nome da seção do artigo em espanhol\']',
+                'advice': None
+                },...
+            ]
+        """
+        obtained_toc_sections = _get_sections(self.article_toc_sections)
+        for lang, sections in expected_toc_sections.items():
+            obtained_toc_section = obtained_toc_sections.get(lang)
+            if obtained_toc_section is not None:
+                is_valid = obtained_toc_section.get('sections') == sections
+                title = obtained_toc_section.get('title')
+                xpath = obtained_toc_section.get('xpath')
+                validation = 'match'
```

Comment:
@Rossi-Luciano a validação do match é outra: validar se o título do documento (article-title) é igual ao título da seção. Neste caso, indicar ERRO. 

Fonte: https://www.scielo.br/media/files/20220900-scielo-brazil-criteria-en.pdf Página 27
Document title different from the section title;

```
Documents with simpler text structures than articles such as editorials, reviews, obituaries, and
letters must also be structured according to the SciELO Publishing Schema↗ and with the mandatory
presence of the following elements:
 - Title of the journal section to which the document belongs;
 - Authorship;
 - Institutional affiliation of the authors;
 - Document title different from the section title;
 - One or more citations in the text; and
 - List of bibliographic references of citations in the body of the text
 ```

Reply:
OK, @robertatakenaka

Reply:
A função de validação é: `validade_article_title_is_different_from_section_titles`
Os testes correspondentes são: `test_validade_article_title_is_different_from_section_titles_success` e `test_validade_article_title_is_different_from_section_titles_fail`

Reply:
@Rossi-Luciano o tipo de validação de `validate_article_toc_sections()` é 'value in list'

Reply:
e o de `validade_article_title_is_different_from_section_titles(self)` é `match`